### ISSUE TITLE
Recursively show status of sync on folders with overlay icon

### ISSUE THREAD

**Issue by ericzolf** [nextcloud/android#3416]
### Is your feature request related to a problem? Please describe.
It is not visible which files might require synchronization and which not, and where they might be. Especially due to issue #2753 it is quite cumbersome to do too often syncs on big folders so that knowing where missing files might be would be very helpful.

### Describe the solution you'd like
Having an overlay icon (the downloaded green arrow or the red cross) on folders like on files, showing recursively if an unsynced file is waiting in the hierarchy to be synced.

### Describe alternatives you've considered
Like there is "All files" and "On Device" have an "Unsynced" view with only the files requiring synchronization. Actually both approaches could be implemented independently from each other.

**Comment by AndyScherzinger** [comment]
Looping in @nextcloud/designers for feedback.

**Comment by jancborchardt** [comment]
> Having an overlay icon (the downloaded green arrow or the red cross) on folders like on files, showing recursively if an unsynced file is waiting in the hierarchy to be synced.

Yeah, the red error cross should propagate to parent folders. Non-error states don’t need to propagate up.

Also, files/folders with error states should be shown on top of any given view. Like favorites, but even above those as errors require action.

> Like there is "All files" and "On Device" have an "Unsynced" view with only the files requiring synchronization. Actually both approaches could be implemented independently from each other.

Yet another view will just make everything less visible. We are a sync app first and foremost, and should ensure that files are synced. Sorting the errors up top will make sure they are seen and you don’t have to navigate to a different view.

**Comment by danielb42** [comment]
> Yeah, the red error cross should propagate to parent folders. Non-error states don’t need to propagate up.

How does this address the initial issue? I'd like to know when there are any unsynced files deep-down in  folders. That is **not** an error-state deserving the red error cross, but also not an okay-state that should go without any info propagation.

If, for any UX-reason beyond me, fully synced folders should not get a green arrow, then let something yellow designate unsynced files..

**Comment by jancborchardt** [comment]
> Wouldn't it be useful to see which folders have been chosen to be synchronized by the user and which not? If I want to check this I have to open every folder and look at the individual files. To me this seems to be a useful basic feature that I have been missing for a long time.

@bpcurse there could be a "partly synced" icon maybe, but that would need the introduction of yet another separate icon. We already have the "On device" view in the sidebar for exactly your usecase.

> If, for any UX-reason beyond me, fully synced folders should not get a green arrow, then maybe we can have something yellow designate "below here be unsynced files"?

@danielb42 folders of which all child folders and files are synced should indeed get a green arrow, didn’t say anything to the contrary. :)

**Comment by bpcurse** [comment]
> folders of which all child folders and files are synced should indeed get a green arrow, didn’t say anything to the contrary. :)

@jancborchardt Thank you for the reply, seems to have been a misunderstanding. Already looking forward to this :+1:

**Comment by tobiasKaminsky** [comment]
> folders of which all child folders and files are synced should indeed get a green arrow, didn’t say anything to the contrary. :)

Currently we do not have such folders. We had this "keep available offline" for files, but this is now replaced by keeping all downloaded files in sync (if possible) on wlan.

**Comment by danielb42** [comment]
@tobiasKaminsky Been a while, but I think the last post still missed the point a bit.

A file downloaded (= _present_ ) on the device has the small green icon (that's good). Now, if *all*  files in folder (and subfolders) are present on the device, then the icon should be shown on the parent folder as well.

I think this has little to do with how "keeping sync" in general is implemented these days. Just that visual indication whether all elements in a folder are effectively present on the device or not, would be so nice..

**Comment by jancborchardt** [comment]
@AndyScherzinger @tobiasKaminsky for the record, fully agree with @danielb42’s comment. :) It’s simply nice visual feedback that everything below that folder is fully synced.

Or is this not something we can assure?

**Comment by tobiasKaminsky** [comment]
> Or is this not something we can assure?

I think this is possible, we then have to store and check each folders eTag and if this matches, check if everything in it is downloaded. And then this recursively, or should the "green icon" only indicate that all files within a folder are downloaded?

**Comment by jancborchardt** [comment]
> And then this recursively, or should the "green icon" only indicate that all files within a folder are downloaded?

Yeah, it should be recursively. That is, green checkmark only if _all_ files and folders below that folder are synced, to be symmetric to what it means in the desktop app.
